internal/server/http/server.go

```diff
@@ -173,6 +184,18 @@ func (s *Server) addImposterHandler(imposters []Imposter) {
 	}
 }
 
-func (s *Server) handleAll(h http.HandlerFunc) {
-	s.router.PathPrefix("/").HandlerFunc(h)
+func (s *Server) handlerMiddleWare(i Imposter) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// handle anything we need before the request is used
+		// We need to grab the request data before the request is used since the body may be empty
+		requestData := GetRequestData(r)
+
+		// handle the imposter reqest
+		handler := ImposterHandler(i)
+		handler(w, r)
```

Comment:
The gorilla/handlers#LoggingHandler does not output the body when it logs, which is what I need/want here far more than the response code. The user should already have access to the response code via the imposters they expect to be returned for the request. So the real improvement here is that I think it will also log for requests that do not have imposters setup. I will instead use the [gorilla/handlers#CustomLoggingHandler](https://pkg.go.dev/github.com/gorilla/handlers#CombinedLoggingHandler) so we get the benefits of it working for all requests but we can also add the body into the log.

Reply:
Wrapped the CustomLoggingHandler with our own CustomLoggingHandler. Unfortunately needed to copy some code from the gorilla/handlers file since they weren't public. Our wrapper handles the logging and writing of the logs to the file regardless of whether there was an imposter setup for it now: https://github.com/friendsofgo/killgrave/pull/170/commits/acaa33c762ae837fa4b5a453fd400e85795b1429

Reply:
I suggest making the logging of the body configurable. In very rare instances, inspecting the body might be necessary. Therefore, it would be more appropriate to enable body logging at level 2 or higher.
> The user should already have access to the response code via the imposters they expect to be returned for the request. 
The same way user can look the body as well from imposter, but thats not the purpose of logging.

Reply:
Before I try to code this I want to be clear with you what is wanted:
- Drop the usage of verbose for the cmd line
- use -l and --log-level for the cmd line for log levels
- log level 0 would be default and not include request logs at all, log level 1 would add in the request logs, level 2 would add the body to the request logs

Is this what you are thinking?

Reply:
Yes, that would be awesome

Reply:
Commit adding in -l --log-level: https://github.com/friendsofgo/killgrave/pull/170/commits/fcc3cb7b5ebcac6feb9193107dd63ca8e9c88e01